### PR TITLE
Adding image generation for azure-openai

### PR DESCRIPTION
**Pull request by edeandrea**
Adding image generation for azure-openai

### CODE REVIEW

File: openai/azure-openai/deployment/src/main/java/io/quarkiverse/langchain4j/azure/openai/deployment/AzureOpenAiProcessor.java [hunk 1]
```diff
@@ -1,8 +1,6 @@
 package io.quarkiverse.langchain4j.azure.openai.deployment;
 
-import static io.quarkiverse.langchain4j.deployment.Langchain4jDotNames.CHAT_MODEL;
-import static io.quarkiverse.langchain4j.deployment.Langchain4jDotNames.EMBEDDING_MODEL;
-import static io.quarkiverse.langchain4j.deployment.Langchain4jDotNames.STREAMING_CHAT_MODEL;
+import static io.quarkiverse.langchain4j.deployment.Langchain4jDotNames.*;
```
Comment (geoand) [inline openai/azure-openai/deployment/src/main/java/io/quarkiverse/langchain4j/azure/openai/deployment/AzureOpenAiProcessor.java:3]: No star imports please

File: openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/AzureOpenAiImageModel.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+                .logRequests(logRequests)
+                .logResponses(logResponses)
+                .build();
+    }
+
+    @Override
+    public Response<Image> generate(String prompt) {
+        GenerateImagesRequest request = requestBuilder(prompt).build();
+
+        GenerateImagesResponse response = withRetry(() -> client.imagesGeneration(request), maxRetries).execute();
```
Comment (geoand) [inline openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/AzureOpenAiImageModel.java:73]: No lambdas please

**Review by geoand — COMMENTED** [review]
Thanks!

I added some small comments

File: openai/azure-openai/deployment/src/main/java/io/quarkiverse/langchain4j/azure/openai/deployment/AzureOpenAiProcessor.java (hunk 1 above)
Comment (edeandrea) [reply openai/azure-openai/deployment/src/main/java/io/quarkiverse/langchain4j/azure/openai/deployment/AzureOpenAiProcessor.java:3]: This was made by your own `impsort` task that I ran on the repo :)

File: openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/AzureOpenAiImageModel.java (hunk 2 above)
Comment (edeandrea) [reply openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/AzureOpenAiImageModel.java:73]: This was copy pasted from your own code in the openai-vanilla module.....

Comment (geoand) [reply openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/AzureOpenAiImageModel.java:73]: Which should be fixed too then 😉

Comment (edeandrea) [reply openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/AzureOpenAiImageModel.java:73]: I'll leave that for another day and another PR. Want to keep this one focused on the task at hand. :)

Comment (geoand) [reply openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/AzureOpenAiImageModel.java:73]: Right, I wasn't proposing to do it now

Comment (edeandrea) [reply openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/AzureOpenAiImageModel.java]: Done

**Review by geoand — APPROVED** [review]
Thanks!

**Pull request merged**